### ISSUE TITLE
ma.masked_equal() change fill_value

### ISSUE THREAD

**Issue by jdemaria** [numpy/numpy#7212]
Hi, trying the example of [(http://docs.scipy.org/doc/numpy-dev/reference/generated/numpy.ma.masked_equal.html#numpy.ma.masked_equal)] I get:
a = np.arange(4)
np.ma.masked_equal(a, 2)
==> masked_array(data = [0 1 -- 3],
                    mask = [False False  True False],
              fill_value = 2)
but
np.ma.masked_where(a == 2, a)
==> masked_array(data = [0 1 -- 3],
                    mask = [False False  True False],
              fill_value = 999999)
tested with 1.10.4

Is it the normal behaviour? if yes so the documentation is wrong and this is not a shortcut of masked_where?

thanks in advance

**Comment by seberg** [comment]
Yes, seems the documentation is incorrect and should be corrected. If you have some time, do you want to create a pull request?

**Comment by jdemaria** [comment]
Note that masked_values() doc is also wrong [http://docs.scipy.org/doc/numpy-dev/reference/generated/numpy.ma.masked_values.html#numpy.ma.masked_values]: "For integers, the fill value will be different in general to the result of masked_equal. + the example"

But I'm a bit troubled by the logic: for me it's strange (and dangerous) to have masked_where() and all masked_xxx() which don't change the fill_value and to have masked_equal() and masked_values() which change it... Sometimes we want to mask values of an array without changing the fill_value, so for these cases I'm forced to use masked_where() or reset the fill_value after masked_equal(), which is tricky. I agree that the current behaviour is also often useful, but this heterogeneous behaviour of masked_xxx() functions seems to me dangerous. And I always felt that mask and fill_value are decoupled in numpy. But maybe I'm completely wrong?
Another idea could be to add a set_fill_value=False option to masked_equal/values(). Or add a fill_value=value.

If you confirm that the current behavior is OK I will try to do a pull request on documentation.

Another question: is there a reason to have a shrink option in masked_values() but not in masked_equal()?

Thanks

**Comment by seberg** [comment]
Frankly, I suggested changing the docs, because I thought it is the
simple solution. I do not use masked arrays, and I am not sure how
confusing the current situation is. But changing the API can be tricky
since you never know who relies on it, warnings are necessary, etc. I
am not against trying/discussing it, it just seems hard on first sight.

We could also add a fill_value keyword argument to some of these
function to make things a bit more obvious. No idea if it would help.
